jwala-webservices/src/main/java/com/cerner/jwala/ws/rest/v1/service/group/impl/GroupServiceRestImpl.java

```diff
@@ -705,9 +711,11 @@ public Response generateAndDeployGroupAppFile(final String groupName, final Stri
                 resourceService.validateSingleResourceForGeneration(resourceIdentifier);
                 if (hostName != null && !hostName.isEmpty()) {
                     // deploy to particular host
+                    historyFacadeService.write(hostName, group, "Deploying application resource "+resourceIdentifier.resourceName+" via operations", EventType.USER_ACTION_INFO, aUser.getUser().getId());
                     performGroupAppDeployToHost(groupName, fileName, appName, hostName, metaData.isHotDeploy());
                 } else {
                     // deploy to all hosts in group
+                    historyFacadeService.write("", group, "Deploying application resource "+resourceIdentifier.resourceName+" via operations to all hosts", EventType.USER_ACTION_INFO, aUser.getUser().getId());
```

Comment:
Again, don't send an empty string for the hostname parameter. You'll have to move this historyFacadeService.write into the performGroupAppDeployToHosts method and get the hostname from each JVM in the group.